### PR TITLE
Feature/interchain txs tests

### PR DESCRIPTION
**Pull request by NeverHappened**
Implements [LSC-115](https://p2pvalidator.atlassian.net/browse/LSC-115)
Added tests for multiple interchain accounts.

Note: to test need to compile updated contracts from [this PR](https://github.com/neutron-org/neutron-contracts/pull/14)

### CODE REVIEW

File: src/testcases/interchaintx.test.ts [hunk 1]
```diff
@@ -29,45 +33,108 @@ describe('Neutron / Interchain TXs', () => {
...
     });
-    test('IBC account created', async () => {
+    test('create ICA1', async () => {
+      const res = await cm.executeContract(
+        contractAddress,
+        JSON.stringify({
+          register: {
+            connection_id: 'connection-0',
+            interchain_account_id: icaId1,
+            denom: 'stake',
```
Comment (swelf19) [inline src/testcases/interchaintx.test.ts:51]: the denom field looks redundant

File: src/testcases/interchaintx.test.ts [hunk 2]
```diff
@@ -76,21 +143,126 @@ describe('Neutron / Interchain TXs', () => {
...
+        {
+          acknowledgement_result: {
+            interchain_account_id: icaId2,
+          },
+        },
+      );
+      expect(res).toMatchObject<AcknowledgementResult>({ error: 'message' });
+    });
+
+    test('delegate undelegate from first ICA, delegate from second ICA', async () => {
```
Comment (swelf19) [inline src/testcases/interchaintx.test.ts:215]: delegate undelegate?

Comment (ratik) [reply src/testcases/interchaintx.test.ts:215]: also what is the idea about undelegating from one and delegating to another?

Comment (NeverHappened) [reply src/testcases/interchaintx.test.ts:215]: Made it just to check that you can simultaneously make different interchain transactions to get different results.
Not sure if it tests something profound

File: src/testcases/interchaintx.test.ts (hunk 1 above)
Comment (oopcode) [reply src/testcases/interchaintx.test.ts:51]: well it doesn't hurt anybody

File: src/testcases/interchaintx.test.ts [hunk 3]
```diff
@@ -29,45 +33,108 @@ describe('Neutron / Interchain TXs', () => {
...
+          acknowledgement_result: { interchain_account_id: icaId2 },
+        },
+      );
+      expect(res2).toBe(null);
+    });
+    test('add some money to ICAs', async () => {
+      const res1 = await cm2.msgSend(icaAddress1.toString(), '10000');
+      expect(res1.code).toEqual(0);
+
+      const res2 = await cm2.msgSend(icaAddress2.toString(), '10000');
```
Comment (sotnikov-s) [inline src/testcases/interchaintx.test.ts:129]: by sending different amounts you can be more assured it all worked as you expected

File: src/testcases/interchaintx.test.ts [hunk 4]
```diff
@@ -20,7 +24,7 @@ describe('Neutron / Interchain TXs', () => {
     cm2 = new CosmosWrapper(testState.sdk_2, testState.wallets.demo2);
   });
 
-  describe('Interchain Tx', () => {
+  describe('Interchain Tx with multiple ICAs', () => {
```
[the remark this answers is not in the document]
Comment (NeverHappened) [reply src/testcases/interchaintx.test.ts:27]: seems good, will do

File: src/testcases/interchaintx.test.ts [hunk 5]
```diff
@@ -29,45 +33,108 @@ describe('Neutron / Interchain TXs', () => {
...
-          port_id: `icacontroller-${contractAddress}.test`,
+          port_id: `icacontroller-${contractAddress}.test1`,
+        },
+        {
+          port_id: `icacontroller-${contractAddress}.test2`,
         },
       ]);
     });
     test('get ica address', async () => {
       await wait(BLOCK_TIME * 20);
```
[the remark this answers is not in the document]
Comment (foxpy) [reply src/testcases/interchaintx.test.ts:83]: I don't think this function would work for this contract, it doesn't implement the query it needs.

[the remark this answers is not in the document]
Comment (sotnikov-s) [reply src/testcases/interchaintx.test.ts:83]: it's an example of how to do the same waiting thing for ica

File: src/testcases/interchaintx.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (NeverHappened) [reply src/testcases/interchaintx.test.ts:28]: Added task for now https://p2pvalidator.atlassian.net/browse/LSC-161

**Comment by oopcode** [comment]
run the tests, everything seems to work fine

**Review by oopcode — DISMISSED** [review]

**Review by sotnikov-s — APPROVED** [review]

**Review by oopcode — APPROVED** [review]

**Pull request merged**